### DISCUSSION TITLE
SSH key connection

### DISCUSSION THREAD

**Discussion by Jasonlovedataanalyst** [CarlosPietrasanta/Tafe230130#1] [General]
Hello Carlos and Sharryn

Thank you for your help of having me in the team
I want to generate a topic of Using SSH keys , there are some benefits of using SSH keys per below: 

1. public is shared between the user and the git repo , and the private key is strictly owned by the private owner. 
2. we do not have to manually input the password , when we push code into the remote repo 

I also have 2 questions , which may need your advice :

1. I know both the information transferred by SSH or HTTPS are both encrypted , do not know whether there are any difference of of the encryption ;
3. whether we have to modify the private key file , so that to make it read only to the owner 

looking forward to your reply 

Best regards ,

Jason

**Comment by CarlosPietrasanta** [comment]
Hi Jason, 

Great topic to discuss. 

I think I can help you with the first question. From my understanding SSH keys are more secure compared to HTTPS (SSH uses a stronger encryption algorithm), and effectively there is no need to input user and password when pushing code as you marked in the second benefit.

In the other hand, If you lose your SSH key passphrase, there's no way to recover it. You'll need to generate a brand new SSH keypair (and ask the repo admin to add the new public key) or switch to HTTPS.

I'm not sure enough to answer your second question, I think it'll better to ask @ozdotcom for some help here.

Thanks!
Carlos.

**Reply by ozdotcom** [reply]
Good Question Jason,

Thanks Carlos for your quick responses to questions.

> "1. I know both the information transferred by SSH or HTTPS are both encrypted , do not know whether there are any difference of of the encryption ; "

What Carlos Said.

> "2. whether we have to modify the private key file , so that to make it read only to the owner:"

I am not convinced we need to change the private key file to read only.

Sharryn
